hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/token/impl/ReadOnlyTokenStore.java

```diff
@@ -27,21 +27,21 @@
 import java.util.Optional;
 
 /**
- * Provides methods for interacting with the underlying data storage mechanisms for working with
- * Tokens.
+ * Provides read-only methods for interacting with the underlying data storage mechanisms for
+ * working with Tokens.
  *
  * <p>This class is not exported from the module. It is an internal implementation detail.
  */
-public class TokenStore {
+public class ReadOnlyTokenStore {
     /** The underlying data storage class that holds the token data. */
     private final State<Long, MerkleToken> tokenState;
 
     /**
-     * Create a new {@link TokenStore} instance.
+     * Create a new {@link ReadOnlyTokenStore} instance.
      *
      * @param states The state to use.
      */
-    public TokenStore(@NonNull final States states) {
+    public ReadOnlyTokenStore(@NonNull final States states) {
```

Comment:
My only thought here is that maybe we should call it a "ReadableTokenStore" instead of a "ReadOnlyTokenStore". The pattern we are using for states was ReadableState and WritableState. A WritableState extends ReadableState. You would say a "Writable state IS A readable state too", but you wouldn't say a "writable state IS A read-only state", since writable and read-only are contradictions.

Same thing here, I'm thinking "ReadableTokenStore" and in the future a "WritableTokenStore extends ReadableTokenStore". What do you think?